### ISSUE TITLE
Get key fails, puppet returns ok.

### ISSUE THREAD

**Issue by networkmaster** [lboynton/puppet-r1soft-cdp-agent#4]
When the get-key command runs, if it fails, puppet reports success

Puppet:
Notice: /Stage[main]/R1soft_cdp_agent/R1soft_cdp_agent::Get_key[http://cdpnode5.xxxx.xxxx.com.au]/Exec[/usr/bin/r1soft-setup --get-key http://cdpnode5.xxxx.xxxx.com.au]/returns: executed successfully

Manually:
[root@monitoring conf]# /usr/bin/r1soft-setup --get-key http://cdpnode5.xxxx.xxxx.com.au
Couldnt connect to remote host
Unable to get key from remote CDP Server
[root@monitoring conf]# host cdpnode5.xxxx.xxxx.com.au
Host cdpnode5.xxxx.xxxx.com.au not found: 3(NXDOMAIN)
[root@monitoring conf]# 

Obviously my issue was a DNS issue, but it does show the issue present with a unhanded error condition

**Comment by lboynton** [comment]
Hmm, perhaps the r1soft-setup command does not set the correct exit code when it can't connect. What does `echo $?` return after running `/usr/bin/r1soft-setup --get-key http://cdpnode5.xxxx.xxxx.com.au` manually when it fails?

**Comment by networkmaster** [comment]
[root@opennmsdev ~]# echo $?
0
[root@opennmsdev ~]# ping 5.5.5.5
PING 5.5.5.5 (5.5.5.5) 56(84) bytes of data.
--- 5.5.5.5 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1000ms
[root@opennmsdev ~]# echo $?
1
[root@opennmsdev ~]# /usr/bin/r1soft-setup --get-key http://cdpnode6.xxxx.xxxx.com.au
Couldnt connect to remote host
Unable to get key from remote CDP Server
[root@opennmsdev ~]# echo $?
0
[root@opennmsdev ~]# 

Guess that's an R1Soft issue not a module issue. I can raise a feature request with R1Soft if you like?

Could get the module to see if the server responds to the R1Soft port prior to doing the key request - might be hard with the limitations of puppet

**Comment by lboynton** [comment]
Yeah, it does look like it's not setting the correct exit code. If R1Soft are open to feature requests then that would probably be the best solution.

**Comment by networkmaster** [comment]
I've opened a feature request.

I agree it's the easiest (and best) way. 

I'll let you know when i get a response
